File: Youtube_python/youtube.py
import os
from datetime import date
from user import User
from premium_user import PremiumUser
from user_bd import UserBd
from video import Video
from video_bd import VideoBd
from channel import Channel
from channel_bd import ChannelBd
from comments import Comments

def menu():
    print("\t \t \t @#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$")
    print("\t \t \t @#$                                                                                 @#$")
    print("\t \t \t @#$                         ~~~ YOUTUBE RECOMMANDER ~~~                             @#$")
    print("\t \t \t @#$                                                                                 @#$")
    print("\t \t \t @#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$")
    print("\n")
    print("\t......................... BIENVENUE SUR LE YOUTUBE RECOMMANDER.!!!!!!!!! ...................")
    print("\n")
    print("\t \t \t************************************************************************************")
    print("\t \t \t*                      MENU DU YOUTUBE RECOMMANDER                                 *")
    print("\t \t \t************************************************************************************")
    print("\n")
    print("\t1............................................  S'inscrire .....................................")
    print("\t2..........................................  Se Connecter .....................................")
    print("\t3........................................  Se deconnecter .....................................")
    print("\t0.............................................. QUITTER  ......................................")

def menu_inscription():
    os.system('cls' if os.name == 'nt' else 'clear')
    print("\n\n\n\n")
    print("1................................ Poster une video...................................")
    print("2.............................   Rechercher une video ...............................")
    print("3................................. Supprimer une video ..............................")
    print("4................................ Creer une chaine ..................................")
    print("5............................. Rechercher une chaine.................................")
    print("6............................... suivre une chaine ..................................")
    print("7............................. Arreter de suivre une chaine .........................")
    print("8.......................... Poster une video dans la chaine .........................")
    print("9.......................... Supprimer une video de la chaine ........................")
    print("10.................................. Retour au menu principal .......................")
    print("11.................................. Afficher mon profil ............................")
    print("0............................. Fermer le Youtube Recommander ........................")

def menu_payment():
    os.system('cls' if os.name == 'nt' else 'clear')
    print("\n\n\n\n")
    print("1........................ S'incrire comme un utilisateur............................")
    print("2................... S'incrire comme un utilisateur premium ........................")
    print("0................................. Retour  ...........................................")

def main():
    video = Video()
    user_bd = UserBd()
    video_bd = VideoBd()
    channel_bd = ChannelBd()
    current_user = None
    
    while True:
        menu()
        try:
            choix = int(input("veuiller entrer une reponse : "))
        except ValueError:
            print("Choix invalide.")
            continue
        
        if choix == 1:  # S'inscrire
            menu_payment()
            try:
                inscrip_choix = int(input("veuiller entrer une reponse : "))
            except ValueError:
                print("Choix invalide.")
                continue
            
            if inscrip_choix == 1:  # Utilisateur standard
                name = input("Entrer votre Nom complet: ").strip()
                gender = input("Entrer votre genre : ")
                num_tel = int(input("Entrer votre numero de telephone : "))
                email = input("Entrer votre adresse mail : ").strip()
                password = input("Entrer votre mot de passe : ").strip()
                
                new_user = User.create_user(name, gender, num_tel, email, password)
                user_bd.add_user(new_user)
                print(f"Inscription réussie! Votre ID est: {new_user.user_id}")
                current_user = new_user
            
            elif inscrip_choix == 2:  # Utilisateur premium
                name = input("Entrer votre Nom complet: ").strip()
                gender = input("Entrer votre genre : ")
                num_tel = int(input("Entrer votre numero de telephone : "))
                email = input("Entrer votre adresse mail : ").strip()
                password = input("Entrer votre mot de passe : ").strip()
                
                new_premium_user = PremiumUser.create_premium_user(name, gender, num_tel, email, password, None)
                new_premium_user.subscription_payment()
                user_bd.add_premium_user(new_premium_user)
                print(f"Inscription Premium réussie! Votre ID est: {new_premium_user.user_id}")
                current_user = new_premium_user
            
            elif inscrip_choix == 0:
                print("Retour au menu principal.....")
                continue
            
            # Menu après inscription
            if current_user:
                handle_user_menu(current_user, video, video_bd, channel_bd, user_bd)
        
        elif choix == 2:  # Se connecter
            logged_user = user_bd.log_in()
            if not logged_user:
                print("\n CONNEXION ÉCHOUÉE ! Retour au menu principal...\n")
                continue
            
            current_user = logged_user
            handle_user_menu(current_user, video, video_bd, channel_bd, user_bd)
        
        elif choix == 3:  # Se déconnecter
            email = input("Entrez l'email du compte: ")
            user_bd.log_out(email)
            current_user = None
            print("Vous êtes déconnecté.")
        
        elif choix == 0:  # Quitter
            print("Merci pour votre visite!")
            break
        
        else:
            print("Choix invalide. Veuillez réessayer.")

def handle_user_menu(current_user, video, video_bd, channel_bd, user_bd):
    while True:
        menu_inscription()
        try:
            reponse = int(input("veuillez entrer un choix :"))
        except ValueError:
            print("Choix invalide. Veuillez entrer un numéro.")
            continue
        
        if reponse == 1:  # Poster une vidéo
            author = input("Entrer votre nom : ")
            titre = input("Entrer le titre de la video :")
            description = input("Entrer la description de la video : ")
            resolution = int(input("Entrer la resolution desiree de votre video (ex: 720, 1080, 4, 8) : "))
            
            new_video = Video.create_video(titre, author, description, resolution, date.today())
            video_bd.add_video(new_video)
            current_user.add_video(new_video)
            
            print(f"Vidéo créée avec l'ID: {new_video.video_id}")
            print("\n")
            video_bd.display_video()
            
            rep = input("\n souhaitez vous liker ?(O/N)...").upper()
            if rep == 'O':
                video_bd.liked_video(titre)
                current_user.add_liked_video(new_video.video_id)
                print("\n")
                video_bd.display_video()
            
            re = input("\n Souhaitez-vous laisser un commentaire ? (O/N)...").upper()
            if re == 'O':
                nme = input("Entrez votre nom:")
                cm = input("Entrez votre commentaire :")
                
                new_comment = Comments.create_comment(nme, date.today(), cm)
                video.add_comment(new_comment)
                current_user.add_comment(new_comment)
                
                print(f"Commentaire créé avec l'ID: {new_comment.comment_id}")
                print("\n")
                video.display_list_comments()
                
                v = input("Souhaitez vous liker un commentaire ?(O/N)...").upper()
                if v == 'O':
                    iden = input("Entrez l'id :")
                    video.liked_comment(iden)
        
        elif reponse == 2:  # Rechercher une vidéo
            titre_recherche = input(" Entrer le titre de la video que vous desirez rechercher :")
            video_bd.research_video(titre_recherche)
            
            ch = input("\n Souhaitez-vous liker cette video ?(O/N)").upper()
            if ch == 'O':
                video_bd.liked_video(titre_recherche)
                found_video = video_bd.get_video_by_title(titre_recherche)
                if found_video:
                    current_user.add_liked_video(found_video.video_id)
                video_bd.display_video()
            
            z = input("\n Souhaitez-vous laisser un commentaire ? (O/N)...").upper()
            if z == 'O':
                nme = input("Entrez votre nom:")
                cm = input("Entrez votre commentaire :")
                
                new_comment = Comments.create_comment(nme, date.today(), cm)
                video.add_comment(new_comment)
                current_user.add_comment(new_comment)
                
                print(f"Commentaire créé avec l'ID: {new_comment.comment_id}")
                print("\n")
                video.display_list_comments()
            
            vv = input("Souhaitez vous liker un commentaire ?(O/N)...").upper()
            if vv == 'O':
                identi = input("Entrez l'id :")
                video.liked_comment(identi)
        
        elif reponse == 3:  # Supprimer une vidéo
            n = input("entrez le nom de la video a supprimer :")
            video_bd.delete_video(n)
            print("OPERATION REUSSIE AVEC SUCCES....")
        
        elif reponse == 4:  # Créer une chaîne
            nom = input("Entrer votre nom :")
            nom_chaine = input("Entrez le nom de votre chaine :")
            
            new_channel = Channel.create_channel(nom, nom_chaine, date.today())
            channel_bd.add_channel(new_channel)
            print(f"BIENVENUE DANS LA CHAINE {nom_chaine}")
        
        elif reponse == 5:  # Rechercher une chaîne
            name_channel = input("Entrez le nom de la chaine :")
            channel_bd.search_channel(name_channel)
        
        elif reponse == 6:  # Suivre une chaîne
            n_channel = input("Entrez le nom de la chaine :")
            channel_bd.search_channel(n_channel)
            c = input("\nSouhaitez-vous suivre cette chaine ?(O/N)").upper()
            if c == 'O':
                channel_bd.follow_channel(n_channel)
        
        elif reponse == 7:  # Arrêter de suivre une chaîne
            n_ch = input("Entrez le nom de la chaine :")
            channel_bd.search_channel(n_ch)
            ct = input("\nSouhaitez-vous arreter de suivre cette chaine ?(O/N)").upper()
            if ct == 'O':
                channel_bd.unfollow_channel(n_ch)
        
        elif reponse == 8:  # Poster une vidéo dans la chaîne
            a = input("Entrer votre nom : ")
            t = input("Entrer le titre de la video :")
            d = input("Entrer la description de la video : ")
            r = int(input("Entrer la resolution desiree de votre video (ex: 720, 1080, 4, 8) : "))
            
            channel_video = Video.create_video(t, a, d, r, date.today())
            channel_bd.posted_video(channel_video)
            current_user.add_video(channel_video)
        
        elif reponse == 9:  # Supprimer une vidéo de la chaîne
            idi = input("Entrer l'ID de la vidéo : ")
            channel_bd.delete_video(idi)
            print("OPERATION REUSSIE AVEC SUCCES....")
        
        elif reponse == 10:  # Retour au menu principal
            print("Retour au menu principal...")
            break
        
        elif reponse == 11:  # Afficher mon profil
            if isinstance(current_user, PremiumUser):
                current_user.display_profile_premium_user()
            else:
                user_bd.display_profile_user(current_user.email)
        
        elif reponse == 0:  # Fermer
            print("Merci pour votre fidelite............")
            exit()
        
        else:
            print("Choix invalide. Veuillez réessayer.")

if __name__ == "__main__":
    main()

File: Youtube_c++/Youtube.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <ctime>
#include <sstream>
#include <iomanip>
#include "IdGenerator.h"
#include "User.h"
#include "PremiumUser.h"
#include "UserBd.h"
#include "Video.h"
#include "VideoBd.h"
#include "Channel.h"
#include "ChannelBd.h"
#include "Comments.h"

void menu() {
    std::cout << "\t \t \t @#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$\n";
    std::cout << "\t \t \t @#$                                                                                 @#$\n";
    std::cout << "\t \t \t @#$                         ~~~ YOUTUBE RECOMMANDER ~~~                             @#$\n";
    std::cout << "\t \t \t @#$                                                                                 @#$\n";
    std::cout << "\t \t \t @#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$@#$\n";
    std::cout << "\n";
    std::cout << "\t......................... BIENVENUE SUR LE YOUTUBE RECOMMANDER.!!!!!!!!! ...................\n";
    std::cout << "\n";
    std::cout << "\t \t \t************************************************************************************\n";
    std::cout << "\t \t \t*                      MENU DU YOUTUBE RECOMMANDER                                 *\n";
    std::cout << "\t \t \t************************************************************************************\n";
    std::cout << "\n";
    std::cout << "\t1............................................  S'inscrire .....................................\n";
    std::cout << "\t2..........................................  Se Connecter .....................................\n";
    std::cout << "\t3........................................  Se deconnecter .....................................\n";
    std::cout << "\t0.............................................. QUITTER  ......................................\n";
}

void menuInscription() {
    #ifdef _WIN32
        system("cls");
    #else
        system("clear");
    #endif
    
    std::cout << "\n\n\n\n";
    std::cout << "1................................ Poster une video...................................\n";
    std::cout << "2.............................   Rechercher une video ...............................\n";
    std::cout << "3................................. Supprimer une video ..............................\n";
    std::cout << "4................................ Creer une chaine ..................................\n";
    std::cout << "5............................. Rechercher une chaine.................................\n";
    std::cout << "6............................... suivre une chaine ..................................\n";
    std::cout << "7............................. Arreter de suivre une chaine .........................\n";
    std::cout << "8.......................... Poster une video dans la chaine .........................\n";
    std::cout << "9.......................... Supprimer une video de la chaine ........................\n";
    std::cout << "10.................................. Retour au menu principal .......................\n";
    std::cout << "11.................................. Afficher mon profil ............................\n";
    std::cout << "0............................. Fermer le Youtube Recommander ........................\n";
}

void menuPayment() {
    #ifdef _WIN32
        system("cls");
    #else
        system("clear");
    #endif
    
    std::cout << "\n\n\n\n";
    std::cout << "1........................ S'incrire comme un utilisateur............................\n";
    std::cout << "2................... S'incrire comme un utilisateur premium ........................\n";
    std::cout << "0................................. Retour  .........................................\n";
}

std::string getCurrentDate() {
    time_t now = time(0);
    tm* ltm = localtime(&now);
    std::ostringstream oss;
    oss << (1900 + ltm->tm_year) << "-" 
        << std::setw(2) << std::setfill('0') << (1 + ltm->tm_mon) << "-" 
        << std::setw(2) << std::setfill('0') << ltm->tm_mday;
    return oss.str();
}

void handleUserMenu(User* currentUser, Video& video, VideoBd& videoBd, ChannelBd& channelBd, UserBd& userBd);

int main() {
    IdGenerator::initialize();
    
    Video video;
    UserBd userBd;
    VideoBd videoBd;
    ChannelBd channelBd;
    User* currentUser = nullptr;
    
    while (true) {
        menu();
        std::cout << "veuiller entrer une reponse : ";
        int choix;
        std::cin >> choix;
        std::cin.ignore();
        
        if (choix == 1) {  // S'inscrire
            menuPayment();
            std::cout << "veuiller entrer une reponse : ";
            int inscripChoix;
            std::cin >> inscripChoix;
            std::cin.ignore();
            
            if (inscripChoix == 1) {  // Utilisateur standard
                std::string name, gender, email, password;
                int numTel;
                
                std::cout << "Entrer votre Nom complet: ";
                std::getline(std::cin, name);
                std::cout << "Entrer votre genre : ";
                std::getline(std::cin, gender);
                std::cout << "Entrer votre numero de telephone : ";
                std::cin >> numTel;
                std::cin.ignore();
                std::cout << "Entrer votre adresse mail : ";
                std::getline(std::cin, email);
                std::cout << "Entrer votre mot de passe : ";
                std::getline(std::cin, password);
                
                User* newUser = new User(User::createUser(name, gender, numTel, email, password));
                userBd.addUser(newUser);
                std::cout << "Inscription réussie! Votre ID est: " << newUser->getUserId() << "\n";
                currentUser = newUser;
                
            } else if (inscripChoix == 2) {  // Utilisateur premium
                std::string name, gender, email, password;
                int numTel;
                
                std::cout << "Entrer votre Nom complet: ";
                std::getline(std::cin, name);
                std::cout << "Entrer votre genre : ";
                std::getline(std::cin, gender);
                std::cout << "Entrer votre numero de telephone : ";
                std::cin >> numTel;
                std::cin.ignore();
                std::cout << "Entrer votre adresse mail : ";
                std::getline(std::cin, email);
                std::cout << "Entrer votre mot de passe : ";
                std::getline(std::cin, password);
                
                PremiumUser* newPremiumUser = new PremiumUser(PremiumUser::createPremiumUser(name, gender, numTel, email, password, ""));
                newPremiumUser->subscriptionPayment();
                userBd.addPremiumUser(newPremiumUser);
                std::cout << "Inscription Premium réussie! Votre ID est: " << newPremiumUser->getUserId() << "\n";
                currentUser = newPremiumUser;
                
            } else if (inscripChoix == 0) {
                std::cout << "Retour au menu principal.....\n";
                continue;
            }
            
            // Menu après inscription
            if (currentUser != nullptr) {
                handleUserMenu(currentUser, video, videoBd, channelBd, userBd);
            }
            
        } else if (choix == 2) {  // Se connecter
            User* loggedUser = userBd.logIn();
            
            if (loggedUser == nullptr) {
                std::cout << "\n CONNEXION ÉCHOUÉE ! Retour au menu principal...\n\n";
                continue;
            }
            
            currentUser = loggedUser;
            handleUserMenu(currentUser, video, videoBd, channelBd, userBd);
            
        } else if (choix == 3) {  // Se déconnecter
            std::string email;
            std::cout << "Entrez l'email du compte: ";
            std::getline(std::cin, email);
            userBd.logOut(email);
            currentUser = nullptr;
            std::cout << "Vous êtes déconnecté.\n";
            
        } else if (choix == 0) {  // Quitter
            std::cout << "Merci pour votre visite!\n";
            break;
            
        } else {
            std::cout << "Choix invalide. Veuillez réessayer.\n";
        }
    }
    
    return 0;
}

void handleUserMenu(User* currentUser, Video& video, VideoBd& videoBd, ChannelBd& channelBd, UserBd& userBd) {
    int reponse;
    do {
        menuInscription();
        std::cout << "veuillez entrer un choix :";
        std::cin >> reponse;
        std::cin.ignore();
        
        if (reponse == 1) {  // Poster une vidéo
            std::string author, titre, description;
            int resolution;
            
            std::cout << "Entrer votre nom : ";
            std::getline(std::cin, author);
            std::cout << "Entrer le titre de la video :";
            std::getline(std::cin, titre);
            std::cout << "Entrer la description de la video : ";
            std::getline(std::cin, description);
            std::cout << "Entrer la resolution desiree de votre video (ex: 720, 1080, 4, 8) : ";
            std::cin >> resolution;
            std::cin.ignore();
            
            Video newVideo = Video::createVideo(titre, author, description, resolution, getCurrentDate());
            videoBd.addVideo(newVideo);
            currentUser->addVideo(newVideo);
            
            std::cout << "Vidéo créée avec l'ID: " << newVideo.getVideoId() << "\n\n";
            videoBd.displayVideo();
            
            std::string rep;
            std::cout << "\n souhaitez vous liker ?(O/N)...";
            std::getline(std::cin, rep);
            if (rep == "O" || rep == "o") {
                videoBd.likedVideo(titre);
                currentUser->addLikedVideo(newVideo.getVideoId());
                std::cout << "\n";
                videoBd.displayVideo();
            }
            
            std::string re;
            std::cout << "\n Souhaitez-vous laisser un commentaire ? (O/N)...";
            std::getline(std::cin, re);
            if (re == "O" || re == "o") {
                std::string nme, cm;
                std::cout << "Entrez votre nom:";
                std::getline(std::cin, nme);
                std::cout << "Entrez votre commentaire :";
                std::getline(std::cin, cm);
                
                Comments newComment = Comments::createComment(nme, getCurrentDate(), cm);
                video.addComment(newComment);
                currentUser->addComment(newComment);
                
                std::cout << "Commentaire créé avec l'ID: " << newComment.getCommentId() << "\n\n";
                video.displayListComments();
                
                std::string v;
                std::cout << "Souhaitez vous liker un commentaire ?(O/N)...";
                std::getline(std::cin, v);
                if (v == "O" || v == "o") {
                    std::string iden;
                    std::cout << "Entrez l'id :";
                    std::getline(std::cin, iden);
                    video.likedComment(iden);
                }
            }
            
        } else if (reponse == 2) {  // Rechercher une vidéo
            std::string titreRecherche;
            std::cout << " Entrer le titre de la video que vous desirez rechercher :";
            std::getline(std::cin, titreRecherche);
            videoBd.researchVideo(titreRecherche);
            
            std::string ch;
            std::cout << "\n Souhaitez-vous liker cette video ?(O/N)";
            std::getline(std::cin, ch);
            if (ch == "O" || ch == "o") {
                videoBd.likedVideo(titreRecherche);
                Video* foundVideo = videoBd.getVideoByTitle(titreRecherche);
                if (foundVideo != nullptr) {
                    currentUser->addLikedVideo(foundVideo->getVideoId());
                }
                videoBd.displayVideo();
            }
            
        } else if (reponse == 3) {  // Supprimer une vidéo
            std::string n;
            std::cout << "entrez le nom de la video a supprimer :";
            std::getline(std::cin, n);
            videoBd.deleteVideo(n);
            std::cout << "OPERATION REUSSIE AVEC SUCCES....\n";
            
        } else if (reponse == 4) {  // Créer une chaîne
            std::string nom, nomChaine;
            std::cout << "Entrer votre nom :";
            std::getline(std::cin, nom);
            std::cout << "Entrez le nom de votre chaine :";
            std::getline(std::cin, nomChaine);
            
            Channel newChannel = Channel::createChannel(nom, nomChaine, getCurrentDate());
            channelBd.addChannel(newChannel);
            std::cout << "BIENVENUE DANS LA CHAINE " << nomChaine << "\n";
            
        } else if (reponse == 5) {  // Rechercher une chaîne
            std::string nameChannel;
            std::cout << "Entrez le nom de la chaine :";
            std::getline(std::cin, nameChannel);
            channelBd.searchChannel(nameChannel);
            
        } else if (reponse == 6) {  // Suivre une chaîne
            std::string nChannel;
            std::cout << "Entrez le nom de la chaine :";
            std::getline(std::cin, nChannel);
            channelBd.searchChannel(nChannel);
            
            std::string c;
            std::cout << "\nSouhaitez-vous suivre cette chaine ?(O/N)";
            std::getline(std::cin, c);
            if (c == "O" || c == "o") {
                channelBd.followChannel(nChannel);
            }
            
        } else if (reponse == 7) {  // Arrêter de suivre une chaîne
            std::string nCh;
            std::cout << "Entrez le nom de la chaine :";
            std::getline(std::cin, nCh);
            channelBd.searchChannel(nCh);
            
            std::string ct;
            std::cout << "\nSouhaitez-vous arreter de suivre cette chaine ?(O/N)";
            std::getline(std::cin, ct);
            if (ct == "O" || ct == "o") {
                channelBd.unfollowChannel(nCh);
            }
            
        } else if (reponse == 8) {  // Poster une vidéo dans la chaîne
            std::string a, t, d;
            int r;
            
            std::cout << "Entrer votre nom : ";
            std::getline(std::cin, a);
            std::cout << "Entrer le titre de la video :";
            std::getline(std::cin, t);
            std::cout << "Entrer la description de la video : ";
            std::getline(std::cin, d);
            std::cout << "Entrer la resolution desiree de votre video (ex: 720, 1080, 4, 8) : ";
            std::cin >> r;
            std::cin.ignore();
            
            Video channelVideo = Video::createVideo(t, a, d, r, getCurrentDate());
            channelBd.postedVideo(channelVideo);
            currentUser->addVideo(channelVideo);
            
        } else if (reponse == 9) {  // Supprimer une vidéo de la chaîne
            std::string idi;
            std::cout << "Entrer l'ID de la vidéo : ";
            std::getline(std::cin, idi);
            channelBd.deleteVideo(idi);
            std::cout << "OPERATION REUSSIE AVEC SUCCES....\n";
            
        } else if (reponse == 10) {  // Retour au menu principal
            std::cout << "Retour au menu principal...\n";
            
        } else if (reponse == 11) {  // Afficher mon profil
            PremiumUser* premiumUser = dynamic_cast<PremiumUser*>(currentUser);
            if (premiumUser != nullptr) {
                premiumUser->displayProfilePremiumUser();
            } else {
                userBd.displayProfileUser(currentUser->getEmail());
            }
            
        } else if (reponse == 0) {  // Fermer
            std::cout << "Merci pour votre fidelite............\n";
            exit(0);
            
        } else {
            std::cout << "Choix invalide. Veuillez réessayer.\n";
        }
        
    } while (reponse != 10 && reponse != 0);
}

/*
COMPILATION AVEC GCC:
=====================

Pour compiler ce projet avec gcc/g++, utilisez les commandes suivantes:

1. Compiler tous les fichiers .cpp:
   g++ -c IdGenerator.cpp Comments.cpp Video.cpp VideoBd.cpp User.cpp PremiumUser.cpp UserBd.cpp Channel.cpp ChannelBd.cpp youtube.cpp

2. Lier tous les fichiers objets:
   g++ IdGenerator.o Comments.o Video.o VideoBd.o User.o PremiumUser.o UserBd.o Channel.o ChannelBd.o youtube.o -o youtube

3. Ou en une seule commande:
   g++ IdGenerator.cpp Comments.cpp Video.cpp VideoBd.cpp User.cpp PremiumUser.cpp UserBd.cpp Channel.cpp ChannelBd.cpp youtube.cpp -o youtube

4. Exécuter le programme:
   ./youtube

STRUCTURE DES FICHIERS:
=======================
IdGenerator.h / IdGenerator.cpp
Comments.h / Comments.cpp
Video.h / Video.cpp
VideoBd.h / VideoBd.cpp
User.h / User.cpp
PremiumUser.h / PremiumUser.cpp
UserBd.h / UserBd.cpp
Channel.h / Channel.cpp
ChannelBd.h / ChannelBd.cpp
youtube.cpp (main)

*/
